lib/generators.nix

```diff
@@ -31,26 +30,28 @@ rec {
    * suitable for bash scripts but not much else.
    */
   mkValueStringDefault = {}: v: with builtins;
-    let err = t: v: abort
-          ("generators.mkValueStringDefault: " +
-           "${t} not supported: ${toPretty {} v}");
-    in   if isInt      v then toString v
-    # we default to not quoting strings
-    else if isString   v then v
-    # isString returns "1", which is not a good default
-    else if true  ==   v then "true"
-    # here it returns to "", which is even less of a good default
-    else if false ==   v then "false"
-    else if null  ==   v then "null"
-    # if you have lists you probably want to replace this
-    else if isList     v then err "lists" v
-    # same as for lists, might want to replace
-    else if isAttrs    v then err "attrsets" v
-    # functions can’t be printed of course
-    else if isFunction v then err "functions" v
-    # let’s not talk about floats. There is no sensible `toString` for them.
-    else if isFloat    v then err "floats" v
-    else err "this value is" (toString v);
+    let
+      err = t: v: abort
+        ("generators.mkValueStringDefault: " + "${t} not supported: ${toPretty { } v}"
+        );
```

Comment:
If binops get wrapped, `)` should also be wrapped because there is no newline after `(`. Or should `(` get expaded?

Reply:
fixed in the new commit.